packages/actor-extract-links-extract-tree/lib/treeMetadataExtraction.ts

```diff
@@ -0,0 +1,86 @@
+import type * as RDF from 'rdf-js';
+import { termToString } from 'rdf-string';
+import { TreeNodes, RelationOperatorReversed } from './TreeMetadata';
+import type { ITreeRelation, ITreeRelationRaw, SparqlRelationOperator } from './TreeMetadata';
+
```

Comment:
Do we really need this separate file here?
Can we move them into `ActorExtractLinksTree`?

Reply:
I think they make there own concepts and I think it would be strange to import the whole actor into the other classes of the package, just for those metadata objects.

Reply:
But all of the functions here are only used inside that actor, so it makes sense to move them there, right?

Reply:
Not they are used by the solver too, and the filter node class